devito/dimension.py

```diff
@@ -0,0 +1,22 @@
+from sympy import Symbol
+
+__all__ = ['Dimension', 'x', 'y', 'z', 't']
+
+
+class Dimension(Symbol):
```

Comment:
Yes, the nomenclature here is debatable, and I'm not insisting on `Dimension` at all. I would argue though that at the top-level, ie in `f(t, x, y)` the dimension does not yet define a loop, and that is the most common place users will define it, so `LoopDimension` or `LoopIndex` seem slightly too specific to me.